src/gmt_init.c

```diff
@@ -10654,11 +10667,24 @@ unsigned int gmtlib_setparameter (struct GMT_CTRL *GMT, const char *keyword, cha
 		case GMTCASE_MAP_HEADING_OFFSET:
 			GMT->current.setting.map_heading_offset = gmt_M_to_inch (GMT, value);
 			break;
+		case GMTCASE_MAP_LABEL_MODE:
+			if ((i = sscanf (value, "%[^/]/%s", txt_a, txt_b)) == 2) {	/* Separate settings for x and y */
+				GMT->current.setting.map_label_mode[GMT_X] = (!strcmp (txt_a, "axis") ? GMT_LABEL_AXIS : GMT_LABEL_ANNOT);
+				GMT->current.setting.map_label_mode[GMT_Y] = (!strcmp (txt_b, "axis") ? GMT_LABEL_AXIS : GMT_LABEL_ANNOT);
+			}
+			else
+				GMT->current.setting.map_label_mode[GMT_X] = GMT->current.setting.map_label_mode[GMT_Y] = (!strcmp (value, "axis") ? GMT_LABEL_AXIS : GMT_LABEL_ANNOT);
+			break;
```

Comment:
It would be good to add a check here that the value is "axis" or "auto" and to report an error if not. Let me know if you want me to do this.

Reply:
Yes, agreed - I was being lazy.  Although you mean axis or annot, right?

Reply:
Yes - clearly I spend too much time thinking about auto settings. axis or annot.